modules/mono/editor/GodotTools/GodotTools/Build/NuGetUtils.cs

```diff
@@ -0,0 +1,296 @@
+            // - Mono/.NETFramework (standalone NuGet):
+            //     Uses Environment.SpecialFolder.ApplicationData
+            //     - Windows: '%APPDATA%'
+            //     - Linux/macOS: '$HOME/.config'
+            // - CoreCLR (dotnet CLI NuGet):
+            //     - Windows: '%APPDATA%'
+            //     - Linux/macOS: '$DOTNET_CLI_HOME/.nuget' otherwise '$HOME/.nuget'
+
+            string applicationData = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
+
+            if (Utils.OS.IsWindows)
+            {
+                // %APPDATA% for both
+                return new[] {Path.Combine(applicationData, "NuGet", "NuGet.Config")};
+            }
+
+            var paths = new string[2];
+
+            // CoreCLR (dotnet CLI NuGet)
+
+            string dotnetCliHome = Environment.GetEnvironmentVariable("DOTNET_CLI_HOME");
+            if (!string.IsNullOrEmpty(dotnetCliHome))
+            {
+                paths[0] = Path.Combine(dotnetCliHome, ".nuget", "NuGet", "NuGet.Config");
+            }
+            else
+            {
+                string home = Environment.GetEnvironmentVariable("HOME");
+                if (string.IsNullOrEmpty(home))
+                    throw new InvalidOperationException("Required environment variable 'HOME' is not set.");
+                paths[0] = Path.Combine(home, ".nuget", "NuGet", "NuGet.Config");
+            }
+
+            // Mono/.NETFramework (standalone NuGet)
+
+            // ApplicationData is $HOME/.config on Linux/macOS
+            paths[1] = Path.Combine(applicationData, "NuGet", "NuGet.Config");
+
+            return paths;
+        }
+
+        // nupkg extraction
+        //
+        // Exclude: (NuGet.Client -> NuGet.Packaging.PackageHelper.ExcludePaths)
+        // package/
+        // _rels/
+        // [Content_Types].xml
+        //
+        // Don't ignore files that begin with a dot (.)
+        //
+        // The nuspec is not lower case inside the nupkg but must be made lower case when extracted.
+
+        /// <summary>
+        /// Adds the specified fallback folder to the user NuGet.Config files,
+        /// for both standalone NuGet (Mono/.NETFramework) and dotnet CLI NuGet.
+        /// </summary>
+        public static void AddFallbackFolderToUserNuGetConfigs(string name, string path)
+        {
+            foreach (string nuGetConfigPath in GetAllUserNuGetConfigFilePaths())
+            {
+                if (!System.IO.File.Exists(nuGetConfigPath))
+                {
+                    // It doesn't exist, so we create a default one
+                    const string defaultConfig = @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+  <packageSources>
+    <add key=""nuget.org"" value=""https://api.nuget.org/v3/index.json"" protocolVersion=""3"" />
+  </packageSources>
+</configuration>
+";
+                    System.IO.File.WriteAllText(nuGetConfigPath, defaultConfig, Encoding.UTF8); // UTF-8 with BOM
```

Comment:
Is a BOM necessary for NuGet to recognize UTF-8? It's best practice to exclude the BOM unless there is [a particular oddity](https://github.com/dotnet/sdk/issues/8929).

Reply:
I checked all the `NuGet.Config` files I had and they include a BOM. I don't know if it's necessary, but considering it's the case for solution files I rather stay safe.